react/components/CSVUploader.tsx

```diff
@@ -23,31 +43,56 @@ interface CSVUploaderProps {
   setMessages: (messages: MessagesOfProvider[] | null) => void
 }
 
+type ErrorCode = null | 'REJECTED' | 'NO_TRANSLATABLE_FIELD_FOUND' | 'ID_NOT_FOUND'
+
+function errorToMessage(
+  error: ErrorCode,
+  { formatMessage }: InjectedIntl
+): string {
+  switch (error) {
+    case null:
+      return ''
+    case 'REJECTED':
+      return formatMessage(notSupportedFileMessage)
+    case 'NO_TRANSLATABLE_FIELD_FOUND':
+      return formatMessage(noTranslatableFieldMessage)
+    case 'ID_NOT_FOUND':
+      return formatMessage(noIdColumnMessage)
+  }
```

Comment:
I never use switch cases in Javascript because it [can easily introduce side effects](https://github.com/dwyl/Javascript-the-Good-Parts-notes#-appendix-b---the-bad-parts).

I suggest always using object notation:

```js
const errorCodes = {
  REJECTED: notSupportedFileMessage,
  NO_TRANSLATABLE_FIELD_FOUND: noTranslatableFieldMessage,
  ID_NOT_FOUND: noIdColumnMessage,
}

return (errorCodes[error] && formatMessage(errorCodes[error]) || ''
```

Reply:
In this case, I think `switch` is cleaner - because we are returning right away after every case.

In general, this issue can be solved easily with proper tooling (https://eslint.org/docs/rules/no-fallthrough).